java/libs/src/main/java/org/physical_web/collection/PwsClient.java

```diff
@@ -79,13 +121,67 @@ private void recordResponse() {
         pwsResultCallback.onResponseReceived(new Date().getTime() - startTime);
       }
 
+      private PwsResult getPwsResult(JSONObject jsonUrlMetadata){
+        switch(apiVersion){
+          case 1:
+            return getV1PwsResult(jsonUrlMetadata);
+          case 2:
+            return getV2PwsResult(jsonUrlMetadata);
+          default:
+            throw new RuntimeException(UKNOWN_API_ERROR_MESSAGE);
+        }
+      }
+
+      private PwsResult getV1PwsResult(JSONObject jsonUrlMetadata){
+        try {
+            return new PwsResult.Builder(
+                jsonUrlMetadata.getString("id"), jsonUrlMetadata.getString("url"))
+                .setTitle(jsonUrlMetadata.optString("title"))
+                .setDescription(jsonUrlMetadata.optString("description"))
+                .setIconUrl(jsonUrlMetadata.optString("icon"))
+                .setGroupId(jsonUrlMetadata.optString("groupId"))
+                .build();
+        } catch (JSONException e) {
+          return null;
+        }
+      }
+
+      private PwsResult getV2PwsResult(JSONObject jsonUrlMetadata){
+        try {
+            JSONObject jsonPageInfo = jsonUrlMetadata.getJSONObject("pageInfo");
+            return new PwsResult.Builder(
+                jsonUrlMetadata.getString("scannedUrl"), jsonUrlMetadata.getString("resolvedUrl"))
+                .setTitle(jsonPageInfo.optString("title"))
+                .setDescription(jsonPageInfo.optString("description"))
+                .setIconUrl(jsonPageInfo.optString("icon"))
+                // TODO(hayesjordan): Add group id when V2 is changed to have group id.
+                .setGroupId("")
```

Comment:
can you leave this unset and correct any errors that get caused by it being null?